source/common/router/rds_impl.cc

```diff
@@ -214,6 +256,21 @@ void RdsRouteConfigProviderImpl::validateConfig(
   ConfigImpl validation_config(config, factory_context_, false);
 }
 
+// Schedules a VHDS request on the main thread and queues up the callback to use when the VHDS
+// response has been propagated to the worker thread that was the request origin.
+bool RdsRouteConfigProviderImpl::requestVirtualHostsUpdate(const std::string& for_domain,
+                                                           std::function<void()> cb) {
+  if (!config()->usesVhds()) {
```

Comment:
This `if` seems a little suspect. It seems like it is checking for a not-really-error but not-really-right condition. The sort of thing where things can muddle along mostly working, and it's hard to track down what's going wrong. Going by the names of the functions involved, I think it might be more safe and solid to, rather than doing this `if` in here, instead guard all uses of `requestVirtualHostsUpdate` with it. I think that would also let you drop the `bool` return type. (If all of that is actually possible).

Reply:
I pulled the check for whether requestVirtualHostsUpdate is available to the very top (see extensions/filters/http/on_demand/on_demand_update.